services/wikiClient.test.js

```diff
@@ -0,0 +1,66 @@
+const axios = require('axios');
+
+const WikiClient = require('./wikiClient');
+
+jest.mock('axios');
+
+describe('WikiClient', () => {
+  it('initializes a new axios client with defaults for header and timeout', () => {
+    const headers = {
+      common: {
+        'Content-Type': 'application/json; charset=utf-8',
+        Accept: 'application/json',
+      },
+    };
+    const timeout = 5000;
+
+    const subject = new WikiClient(); // eslint-disable-line no-unused-vars
+
+    expect(axios.create).toHaveBeenCalledWith({ headers, timeout });
+  });
+
+  describe('getResultsFromApi()', () => {
+    const axiosClient = { get: jest.fn() };
+    const wikiUrl = 'https://en.wikipedia.org';
+    const apiUrl = 'https://en.wikipedia.org/w/api.php';
+    const defaultParams = { action: 'query', format: 'json' };
+    const response = { data: { query: { foo: 'bar' } } };
+
+    describe('when querying for images of a page', () => {
+      const titles = 'Def_Leppard';
+      const prop = 'image';
+      const params = { ...defaultParams, prop, titles };
+
+      it('calls the respective api and returs the query result', async () => {
+        axiosClient.get.mockResolvedValue(response);
+        axios.create.mockReturnValue(axiosClient);
```

Comment:
Can you move the mocking part into a `beforeEach` block? This way the contents of the `it`-block focus on the action and not so much on the setup.